v-next/hardhat/src/internal/builtin-plugins/network-manager/edr/edr-provider.ts

```diff
@@ -397,30 +407,56 @@ async function getProviderConfig(
     networkConfig.chainType,
   );
 
-  const genesisState =
+  const ownedAccounts = await hardhatAccountsToEdrOwnedAccounts(
+    networkConfig.accounts,
+  );
+
+  const genesisState: Map<Uint8Array, AccountOverride> = new Map(
+    ownedAccounts.map(({ secretKey, balance }) => {
+      const address = hexToBytes(addr.parse(secretKey).data);
+      const accountOverride: AccountOverride = {
+        address,
+        balance: BigInt(balance),
+      };
+
+      return [address, accountOverride];
+    }),
+  );
+
+  const chainGenesisState =
     networkConfig.forking !== undefined
       ? [] // TODO: Add support for overriding remote fork state when the local fork is different
       : networkConfig.chainType === "optimism"
         ? opGenesisState(opHardforkFromString(specId))
         : l1GenesisState(l1HardforkFromString(specId));
 
+  for (const account of chainGenesisState) {
+    const existingOverride = genesisState.get(account.address);
+    if (existingOverride !== undefined) {
+      // Favor the genesis state specified by the user
+      account.balance = account.balance ?? existingOverride.balance;
+      account.nonce = account.nonce ?? existingOverride.nonce;
+      account.code = account.code ?? existingOverride.code;
+      account.storage = account.storage ?? existingOverride.storage;
+    } else {
+      genesisState.set(account.address, account);
+    }
+  }
+
   return {
     allowBlocksWithSameTimestamp: networkConfig.allowBlocksWithSameTimestamp,
     allowUnlimitedContractSize: networkConfig.allowUnlimitedContractSize,
     bailOnCallFailure: networkConfig.throwOnCallFailures,
     bailOnTransactionFailure: networkConfig.throwOnTransactionFailures,
     blockGasLimit: networkConfig.blockGasLimit,
-    cacheDir: networkConfig.forking?.cacheDir,
     chainId: BigInt(networkConfig.chainId),
-    chains: hardhatChainDescriptorsToEdrChains(
+    chainOverrides: hardhatChainDescriptorsToEdrChains(
```

Comment:
Can you rename this to `hardhatChainDescriptorsToEdrChainOverrides`?

Reply:
Addressed in https://github.com/NomicFoundation/hardhat/pull/6718/commits/2934961784310b06959c784a4ae060f3a6afccf4